solidity/SolidityParser.g4

```diff
@@ -0,0 +1,861 @@
+// Copyright 2016-2017 Federico Bond <federicobond@gmail.com>
+// Copyright 2024 Dmitry Litovchenko <i@dlitovchenko.ru>
+// 
+// This program is free software: you can redistribute it and/or modify it under the terms of the
+// GNU General Public License as published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+// 
+// This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+// even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+// General Public License for more details.
+// 
+// You should have received a copy of the GNU General Public License along with this program. If
+// not, see <http://www.gnu.org/licenses/>. Copied from https://github.com/ethereum/solidity/blob/60cea4d70012332adcfdcedd8b4f53f556405841/docs/grammar/SolidityParser.g4
+
+// $antlr-format alignTrailingComments true, columnLimit 150, minEmptyLines 1, maxEmptyLinesToKeep 1, reflowComments false, useTab false
+// $antlr-format allowShortRulesOnASingleLine false, allowShortBlocksOnASingleLine true, alignSemicolons hanging, alignColons hanging
+
+/**
+ * Solidity is a statically typed, contract-oriented, high-level language for implementing smart contracts on the Ethereum platform.
+ */
+parser grammar SolidityParser;
```

Comment:
For now i just updated to the last version, since previous was added in 2017, and most of breaking changes added before 2021 and there is no back compatibility between versions. For the future breaking changes it makes sense to have separate grammars